### PR TITLE
Extend batch operations with consistency level

### PR DESCRIPTION
**Pull request by redouan-rhazouani**

### CODE REVIEW

File: weaviate/batch/crud_batch.py [hunk 1]
```diff
@@ -602,12 +608,19 @@ def _create_data(
         weaviate.UnexpectedStatusCodeException
             If weaviate reports a none OK status.
         """
+
+        params = None
+        if self._consistency_level is not None:
+            params = {"consistency_level": self._consistency_level}
```
Comment (dirkkul) [inline weaviate/batch/crud_batch.py:614]: just as a hint, you can do
`param = None if self._consistency_level is None else {"consistency_level": self._consistency_level}`

no need to change it now

File: weaviate/batch/crud_batch.py [hunk 2]
```diff
@@ -1456,6 +1470,14 @@ def dynamic(self, value: bool) -> None:
             self._batching_type = "fixed"
         self._auto_create()
 
+    @property
+    def consistency_level(self, value: Union[ConsistencyLevel, None]) -> Union[str, None]:
```
Comment (dirkkul) [inline weaviate/batch/crud_batch.py:1474]: Optional please :)

File: weaviate/batch/crud_batch.py [hunk 3]
```diff
@@ -1456,6 +1470,14 @@ def dynamic(self, value: bool) -> None:
...
+        return self._consistency_level
+
+    @consistency_level.setter
+    def consistency_level(self, x: Union[ConsistencyLevel, None]) -> None:
```
Comment (dirkkul) [inline weaviate/batch/crud_batch.py:1478]: also optional

**Review by dirkkul — APPROVED** [review]

File: weaviate/batch/crud_batch.py (hunk 3 above)
Comment (redouan-rhazouani) [reply weaviate/batch/crud_batch.py]: Done

File: weaviate/batch/crud_batch.py (hunk 2 above)
Comment (redouan-rhazouani) [reply weaviate/batch/crud_batch.py]: Sure

File: weaviate/batch/crud_batch.py (hunk 1 above)
Comment (redouan-rhazouani) [reply weaviate/batch/crud_batch.py]: here is a shorter one `self._consistency_level = ConsistencyLevel(x).value if x else None`

**Pull request merged**